### PR TITLE
Sof 2071/replace legacy angular material modules

### PR DESCRIPTION
**Pull request by LindvedKrvang**
In Angular v16 the version AngularMaterials that we were using was deprecated.
This PR removes a lot of the AngularMaterial dependencies since we simply don't need them.
We now only depend on the DialogModule of Angular Materiale.

### CODE REVIEW

File: src/app/shared/components/confirmation-dialog/confirmation-dialog.component.scss [hunk 1]
```diff
@@ -1,58 +1,49 @@
...
-    margin-top: 12px;
-    margin-bottom: -12px;
-}
 
-.u-row {
-    display: flex;
-    justify-content: space-between;
-    align-items: center;
-}
+    .dialog-content {
```
Comment (MadsThy) [inline src/app/shared/components/confirmation-dialog/confirmation-dialog.component.scss:39]: i think something is wrong with the indentation in this file :)

File: src/app/shared/components/confirmation-dialog/confirmation-dialog.component.ts [hunk 2]
```diff
@@ -65,4 +72,15 @@ export class ConfirmationDialogComponent extends StronglyTypedDialog<Confirmatio
...
+
+  public confirmDialog(): void {
+    this.dialogRef.close(true)
+  }
+
+  public cancelDialog(): void {
+    this.dialogRef.close(false)
+  }
+
+  protected readonly Icon = Icon
```
Comment (MadsThy) [inline src/app/shared/components/confirmation-dialog/confirmation-dialog.component.ts:84]: should these not be at the top of the file? :)

**Review by MadsThy — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

File: src/app/shared/components/confirmation-dialog/confirmation-dialog.component.scss (hunk 1 above)
Comment (LindvedKrvang) [reply src/app/shared/components/confirmation-dialog/confirmation-dialog.component.scss:39]: Here is a picture of it. It looks good to me
image

File: src/app/shared/components/confirmation-dialog/confirmation-dialog.component.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/app/shared/components/confirmation-dialog/confirmation-dialog.component.ts:84]: They should. It's Webstorm that autoplaced them down there.
I'll move them up in SOF-2364

**Pull request merged**